src/Analyser/AnalysedFilesResolver.php

```diff
@@ -0,0 +1,47 @@
+<?php declare(strict_types = 1);
+
+namespace PHPStan\Analyser;
+
+use function array_fill_keys;
+use function array_map;
+use function strtolower;
+use const PHP_OS_FAMILY;
+
+final class AnalysedFilesResolver
+{
+
+	/** @var bool[] filePath(string) => bool(true) */
+	private array $analysedFiles;
+
+	private bool $caseInsensitiveFilesystem;
+
+	/**
+	 * @param string[] $files
+	 */
+	public function __construct(array $files = [])
+	{
+		$this->caseInsensitiveFilesystem = PHP_OS_FAMILY === 'Darwin';
```

Comment:
instead of hardcoding this for macOS, we could also use a feature detect (which requires a bit of IO)